### PR TITLE
PHOENIX-6995 HA client connections ignore additional jdbc params in t…

### PR DESCRIPTION
**Pull request by tkhurana**
…he jdbc string

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/jdbc/HighAvailabilityGroup.java [hunk 1]
```diff
@@ -770,6 +786,28 @@ public String getUrl2() {
...
+            sb.append(zkUrl);
+            if (!Strings.isNullOrEmpty(additionalJDBCParams)) {
+                sb.append(PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR);
+                sb.append(additionalJDBCParams);
+            }
+            return sb.toString();
+        }
+
+        public String getJDBCUrl1() {
+            return getJDBCUrl(urls.getFirst());
```
Comment (virajjasani) [inline phoenix-core/src/main/java/org/apache/phoenix/jdbc/HighAvailabilityGroup.java:804]: nit: `getJDBCUrl(getUrl1())`

File: phoenix-core/src/main/java/org/apache/phoenix/jdbc/HighAvailabilityGroup.java [hunk 2]
```diff
@@ -770,6 +786,28 @@ public String getUrl2() {
...
+        }
+
+        public String getJDBCUrl2() {
+            return getJDBCUrl(urls.getSecond());
```
Comment (virajjasani) [inline phoenix-core/src/main/java/org/apache/phoenix/jdbc/HighAvailabilityGroup.java:808]: nit: `getJDBCUrl(getUrl2())`

File: phoenix-core/src/main/java/org/apache/phoenix/util/JDBCUtil.java [hunk 3]
```diff
@@ -219,7 +221,10 @@ public static String formatZookeeperUrl(String zkUrl){
         Preconditions.checkArgument(hosts.length > 0,"Unexpected zk url format no hosts found.");
         String hostsStrings = Arrays.stream(hosts).sorted().collect(Collectors.joining(","));
         components[0] = hostsStrings;
-        return Arrays.stream(components).collect(Collectors.joining(":"));
+        // host:port:path:principal
+        // additional arguments passed in url, strip them out
+        int endIdx = Integer.min(components.length, 3);
```
Comment (virajjasani) [inline phoenix-core/src/main/java/org/apache/phoenix/util/JDBCUtil.java:226]: nit: i wonder if we should also make initial validation check for len: 3 or 4
```
Preconditions.checkArgument(components.length == 3 || components.length == 4, "Unexpected zk url format.");
```

**Review by virajjasani — APPROVED** [review]

File: phoenix-core/src/main/java/org/apache/phoenix/util/JDBCUtil.java (hunk 3 above)
Comment (tkhurana) [reply phoenix-core/src/main/java/org/apache/phoenix/util/JDBCUtil.java:226]: Actually, the number of components can be greater than 4 also

**Pull request merged**